packages/shorebird_cli/lib/src/commands/release_command.dart

```diff
@@ -82,7 +82,13 @@ make smaller updates to your app.
       return ExitCode.noUser.code;
     }
 
-    await logValidationIssues();
+    final validationIssues = await logValidationIssues();
+    if (validationIssues.isNotEmpty) {
```

Comment:
I don't think this is exactly what we want because in this case the only check we care about is the network permission check but `logValidationIssues` performs other checks as well (and that list will likely only continue to grow in the future). Tagging @bryanoltman since he worked on the validators.

Reply:
Just a thought Should we return Map<Validator, ValidationIssue> from this method ?
Then here we can easily check for Specific validator.

Reply:
If we want to map validators to the issues they produce, the return type would be `Map<Validator, List<ValidationIssue>>`, as `validate` produces a list of `ValidationIssue`s.

That said, I think we may want do perform this check differently for `release` than for other commands, for a couple reasons:

1) Not all issues should necessarily block a release (we can probably check `validationIssue.severity` and only block on errors)
2) `AndroidInternetPermissionValidator` checks *all* AndroidManifest.xml files, while the release command should only care about `android/app/src/main/AndroidManifest.xml`. It won't affect a release if, for example, `android/app/src/debug/AndroidManifest.xml` is missing the internet permission.

One approach:

1. Separate the validator execution from the logging of validation issues
1. Update `logValidationIssues` to accept a `List<ValidationIssue>` or `Map<Validator, List<ValidationIssues>>`
1. Add some logic, probably per-command, that determines whether to treat `ValidationIssues` as blockers. 
1. Create a new android manifest validator (or modify the existing one) to only check the manifest xml file used for releases.

Reply:
Great idea. will add a property on all commands called `blockOnValidationIssues`.
For manifest validator, will try to modify first, if not then create validator of it.

Reply:
@bryanoltman Few things I have done

- Separated logic for logging validation issue and getting validation issue
- logValidationsIssues takes Map<Validator, List<ValidationIssues>>
- Add `blockOnValidationIssues` flag on shorebird command and getCriticalIssuesCount method
```
    final criticalIssues = await logAndGetCriticalIssueCount();
    if (criticalIssues > 0 && blockOnValidationIssues) {
      logger.err(
        '''Shorebird $name cannot continue until all issues are fixed.''',
      );
      return ExitCode.config.code;
    }
```
- For release, I have added `ReleaseAndroidInternetPermissionValidator` and in release command we are checking like this
```
 final validations = await logAndGetValidationIssues();
    if (validations.isNotEmpty && blockOnValidationIssues) {
      for (final validation in validations.entries) {
        if (validation.key is ReleaseAndroidInternetPermissionValidator) {
          logger.err(
            '''Shorebird $name cannot continue until all issues are fixed.''',
          );
          return ExitCode.config.code;
        }
      }
    }
```

For some reason, last part doesn't look good to me. Can you suggest how to improve this ?

Reply:
After thinking about it some more, I decided to go a slightly different direction with https://github.com/shorebirdtech/shorebird/pull/469. Instead of trying to be too specific about which combinations of validators and issues are blocking, I've changed the logic to block if issues with a severity level of `error` are found. Thanks for your help! I've added you as a coauthor to the new PR.